Python/ceval.c

```diff
@@ -1781,11 +1780,13 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
         }
 
     tracing_dispatch:
+    {
+        int instr_prev = f->f_lasti;
         f->f_lasti = INSTR_OFFSET();
         NEXTOPARG();
 
         if (PyDTrace_LINE_ENABLED())
-            maybe_dtrace_line(f, &trace_info);
+            maybe_dtrace_line(f, &trace_info, instr_prev);
```

Comment:
Why you need to propagate `instr_prev` when `maybe_dtrace_line` already knows about the frame `f`. Cannot you just get it from the frame when you need it?